pkg/operator/ceph/object/spec.go

```diff
@@ -34,21 +34,38 @@ import (
 
 func (c *config) startDeployment() error {
 
-	deployment := &extensions.Deployment{
+	d := &extensions.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      c.instanceName(),
 			Namespace: c.store.Namespace,
 		},
 		Spec: extensions.DeploymentSpec{Template: c.makeRGWPodSpec(), Replicas: &c.store.Spec.Gateway.Instances},
 	}
-	k8sutil.SetOwnerRefs(c.context.Clientset, c.store.Namespace, &deployment.ObjectMeta, c.ownerRefs)
-	_, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.store.Namespace).Create(deployment)
-	return err
+	k8sutil.SetOwnerRefs(c.context.Clientset, c.store.Namespace, &d.ObjectMeta, c.ownerRefs)
+
+	logger.Debugf("starting mds deployment: %+v", d)
+	_, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.store.Namespace).Create(d)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create rgw deployment %s: %+v", c.instanceName(), err)
+		}
+		logger.Infof("deployment for rgw %s already exists. updating if needed", c.instanceName())
+		// There may be a *lot* of rgws, and they are stateless, so don't bother waiting until the
+		// entire deployment is updated to move on.
+		// TODO: is the above statement safe to assume?
+		// TODO: Are there any steps for RGW that need to happen before the daemons upgrade?
```

Comment:
Maybe at one point there are some steps needed. I wouldn't have a problem leaving a comment "here" to point out "add pre update steps of rgw Deployments here or so.